core/modules/server/server.js

```diff
@@ -322,8 +326,16 @@ Server.prototype.listen = function(port,host,prefix) {
 		port = process.env[port] || 8080;
 	}
 	// Warn if required plugins are missing
-	if(!this.wiki.getTiddler("$:/plugins/tiddlywiki/tiddlyweb") || !this.wiki.getTiddler("$:/plugins/tiddlywiki/filesystem")) {
-		$tw.utils.warning("Warning: Plugins required for client-server operation (\"tiddlywiki/filesystem\" and \"tiddlywiki/tiddlyweb\") are missing from tiddlywiki.info file");
+	var missing = [];
+	for (let index = 0; index < this.variables["required-plugins"].length; index++) {
+		const r = this.variables["required-plugins"][index];
+		if (!this.wiki.getTiddler("$:/plugins/"+r)) {
+			missing.push(r);
+		}
+	}
+	if(missing.length > 0) {
+		var error = "Error: Plugin(s) required for client-server operation are missing from the command line or the tiddlywiki.info file: \"" + missing.join("\", \"") + "\"";
+		$tw.utils.error(error);
```

Comment:
I'm not happy with the line length here, but I also don't have a good result. ... What about: 

```
	$tw.utils.error( ["Error: Plugin(s) required for client-server operation are missing from the command line ",
		"or the tiddlywiki.info file: \"", missing.join("\", \""), "\""].join(),
```

It's still "kind of ugly", but would be shorter. @Jermolene ... what do you think

Reply:
Tried a multi-line error.